### PR TITLE
Implement benchmark

### PR DESCRIPTION
**Pull request by danthe96**
- Added benchmark that tests `int8`, `int16`, `int32` and `int64` on lots of sizes (`string` also possible but commented out)
- Added small Python script to generate plot of benchmark
- Uploaded time.png and bandwidth.png as example plots

To run the whole thing, all we need is:
```
benchmark.exe > benchmark.csv
python3 generate_figures.py benchmark.csv
```

I was unsure how row-/column-based layout is supposed to be handled. We're supposed to map depending on the attribute vector size, but that's always just one column. Should we create a square matrix, and then compare the length of one column? Or a fixed number of columns? I can easily extend the benchmark once that's clear.

### CODE REVIEW

**Review by jonaschn — COMMENTED** [review]
Did you intentionally commit the IDE specific files?

**Comment by danthe96** [comment]
Yes, I thought that makes setup easier. Or should I remove it?

File: benchmark/main.cpp [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+static int ITERATIONS = 3;
+
+template <class T>
+long long int benchmark(long col_size, T default_value) {
+    std::vector<T> attribute_vector(col_size / sizeof(T), default_value);
+
+    // Average multiple runs
+    std::vector<long long int> times;
+    for (int i = 0; i < ITERATIONS; i++) {
+        auto start = std::chrono::steady_clock::now();
```
Comment (jonaschn) [inline benchmark/main.cpp]: best practice: use high_resolution_clock instead

File: benchmark/main.cpp [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        for (int j = 0;j < attribute_vector.size(); j++){
+            volatile auto o3_trick = attribute_vector[j];
+        }
+        auto time = std::chrono::duration_cast<std::chrono::nanoseconds>(std::chrono::steady_clock::now() - start);
```
Comment (jonaschn) [inline benchmark/main.cpp]: I am not sure if inlining the clock measurement is equivalent to:
```cpp
auto time_start = std::chrono::high_resolution_clock::now();
<benchmarked operation>
auto time_end = std::chrono::high_resolution_clock::now();
auto time_diff = std::chrono::duration_cast<std::chrono::nanoseconds>(time_end - time_start);
```

**Review by jonaschn — CHANGES_REQUESTED** [review]

**Pull request merged**